src/client/utils/origin-location.js

```diff
@@ -0,0 +1,118 @@
+import sharedUrlUtils from '../../utils/url';
+import { isIframeWithoutSrc } from './dom';
+import { get as getSettings } from '../settings';
+
+const DOCUMENT_URL_RESOLVER = 'hammerhead|document-url-resolver';
+
+class OriginLocation {
+    constructor (document) {
+        //URL processing funcs
+        document[DOCUMENT_URL_RESOLVER] = document.createElement('a');
```

Comment:
Why `urlResolver` (which more likely used by other parts of client code) is exposed by this utility? Why this utility is even class instead of the method set?